Kconfig.zephyr

```
@@ -292,6 +292,17 @@ config LINKER_LAST_SECTION_ID_PATTERN
 	  be used.
 	  The size of the pattern must not exceed 4 bytes.
 
+config LINKER_USE_RELAX
+	bool "Use --relax Linker option"
+	default y
+	help
+	  This option performs global optimizations that become possible when the linker resolves
+	  addressing in the program, such as relaxing address modes and synthesizing new
+	  instructions in the output object file. On platforms where this is not supported,
+	  `--relax' is accepted, but ignored.
+	  Disabling it can reduce performance, as the linker is no longer able to substiture long /
+	  in-effective jump calls to shorter / more effective instructions.
+
```

We need a corresponding `LINKER_USE_NO_RELAX` setting.

This allows other settings to either enforce `relax` (removing the option of disabling it), as well as allowing setting to force disabling of the setting.

```suggestion

config LINKER_USE_NO_RELAX
	bool
	help
	  Hidden symbol to allow features to force the use of no relax.

```